src/Common/src/System/Security/Cryptography/RSACng.cs

```diff
@@ -55,6 +66,13 @@ protected override byte[] HashData(Stream data, HashAlgorithmName hashAlgorithm)
         {
             return CngCommon.HashData(data, hashAlgorithm);
         }
+
+        private void SetKeySize(int newKeySize)
+        {
+            _skipKeySizeCheck = true;
+            KeySize = newKeySize;
```

Comment:
It should be that the only reason it can throw is the key size not being valid (which _skip) is suppressing; but it's a good defensive change.